sunpy/map/mapbase.py

```diff
@@ -684,12 +684,11 @@ def max(self, *args, **kwargs):
 
     @staticmethod
     def _parse_fits_unit(unit_str):
-        unit_str = unit_str.lower()
         replacements = {'gauss': 'G',
                         'dn': 'ct',
                         'dn/s': 'ct/s'}
-        if unit_str in replacements:
-            unit_str = replacements[unit_str]
+        if unit_str.lower() in replacements:
```

Comment:
wait isn't this and the unit test testing if "G" is in the keys of this dict?

Reply:
no?

Reply:
yeah your test isn't testing this code path:

```
>>> u.Unit("G", format="fits")
Unit("G")
 
>>> 'G' in {'gauss': 'G'}
False
```

Reply:
Yes. Thats how it work.